csrc/moe/marlin_kernels/marlin_moe_kernel_ku8.cu

```diff
@@ -0,0 +1,31 @@
+#include "marlin_moe_kernel_ku8.h"
+
+namespace marlin_moe {
+
+// We return bool so we can create these different kernel calls as a sequence
+// of if-elseif's.
+bool call_marlin_moe_kernel_ku8(
+    vllm::ScalarType const& q_type, int thread_n_blocks, int thread_k_blocks,
+    bool has_act_order, int group_blocks, int num_threads, int blocks,
+    int max_shared_mem, cudaStream_t stream, const int4* A_ptr,
+    const int4* B_ptr, int4* C_ptr, const int* sorted_ids_ptr,
+    const float* topk_weights_ptr, const int4* s_ptr, const int4* zp_ptr,
+    const int* g_idx_ptr, int* expert_offsets_ptr, int num_groups,
+    int expert_idx, int num_experts, int topk, int prob_m, int prob_n,
+    int prob_k, int tot_m, int* locks, bool replicate_input, bool apply_weights,
+    int m_block, int max_par, int cfg_max_m_blocks) {
+  bool has_zp = true;
+
+  if (false) {
+  }
+  AWQ_CALL_IF_MOE(vllm::kU8, 16, 4, 256)
+  AWQ_CALL_IF_MOE(vllm::kU8, 8, 8, 256)
+  AWQ_CALL_IF_MOE(vllm::kU8, 8, 4, 128)
+  AWQ_CALL_IF_MOE(vllm::kU8, 4, 8, 128)
```

Comment:
Are there 8bit AWQ models out there? It was my understanding that AutoAWQ only produced 4bit models, so maybe we could save space by dropping 8bit. I'm okay deciding this later

Reply:
I've provided the same type coverage as the unfused Marlin kernel, but I'm ok with removing these if they turn out not to be needed

Reply:
I've talked with @dsikka and we decided to remove the 8-bit zero point code for now. It should be straightforward to add back if we ever need it.

Reply:
Yes, AWQ can quantize to 8-bit. However, for historical reasons we have not done this because there has been no kernel available for execution.